src/components/ActivityFeed/ActivityFeedItem.js

```diff
@@ -0,0 +1,530 @@
+    switch (event.type) {
+      case USER_EVENT:
+        return this.getUserEventData(event);
+      case TRANSACTION_EVENT:
+        return this.getTransactionEventData(event);
+      case COLLECTIBLE_TRANSACTION:
+        return this.getCollectibleTransactionEventData(event);
+      case BADGE_REWARD_EVENT:
+        return this.getBadgeRewardEventData(event);
+      case TYPE_SENT:
+      case TYPE_RECEIVED:
+      case TYPE_ACCEPTED:
+        return this.getSocialEventData(event);
+      default:
+        return null;
+    }
+  }
+
+  renderSettlement = (event: Object) => {
+    const {
+      assets, supportedAssets, asset, feedType, contacts, contactsSmartAddresses, selectEvent,
+    } = this.props;
+    const { type } = event;
+    const trxData = {
+      hideAmount: true,
+      hideSender: true,
+      txType: 'PLR Network settle',
+    };
+
+    const address = this.isReceived(event) ? event.from : event.to;
+    const contact = findMatchingContact(address, contacts, contactsSmartAddresses) || {};
+    const { decimals = 18 } = getAssetData(assets, supportedAssets, event.asset);
+    const value = formatUnits(event.value, decimals);
+
+    return (
+      <SettlementItem
+        settleData={event.extra}
+        onPress={() => selectEvent({
+          ...event,
+          value,
+          contact,
+          ...trxData,
+        }, type, event)}
+        type={feedType}
+        asset={asset}
+        isPending={event === TX_PENDING_STATUS}
+        supportedAssets={supportedAssets}
+        accountAssets={assets}
+      />
+    );
+  }
+
+  getColor = (color: ?string): ?string => {
+    if (!color) return null;
+    const { theme } = this.props;
+    const colors = getThemeColors(theme);
+    return colors[color] || color;
+  }
+
+  render() {
+    const { event, selectEvent } = this.props;
+
+    if (event.type === TRANSACTION_EVENT && event.tag === PAYMENT_NETWORK_TX_SETTLEMENT) {
+      return this.renderSettlement(event);
+    }
+
+    const itemData = this.getEventData(event);
+
+    if (!itemData) return null;
+
+    const {
+      name,
+      icon,
+      status,
+      statusColor,
+      badge,
+      information,
+      statusAsButton,
+      rejectInvitation,
+      acceptInvitation,
+      avatarUrl,
+      imageUrl,
+      iconName,
+      iconColor,
+      itemValue,
+      valueColor,
+      customAddon,
+      statusIcon,
+      eventData,
+    } = itemData;
+
+    return (
+      <ListItemWithImage
+        label={name}
+        subtext={information}
+        onPress={eventData && (() => selectEvent(eventData, event.type, event.status))}
+        badge={badge}
+        actionLabel={status}
+        actionLabelColor={this.getColor(statusColor)}
+        labelAsButton={statusAsButton}
+        rejectInvitation={rejectInvitation}
+        acceptInvitation={acceptInvitation}
+        itemImageSource={icon}
+        avatarUrl={avatarUrl}
+        itemImageUrl={imageUrl}
```

Comment:
It would be good to add `fallbackToGenericToken` for transaction events, so if asset or collectible icon is missing, a generic token would be rendered (rather than "initials").